app/server/appsmith-server/src/main/java/com/appsmith/server/domains/Application.java

```diff
@@ -238,4 +240,16 @@ public enum Type {
             FLUID,
         }
     }
+
+    /**
+     * EmbedSetting is used for embedding Appsmith apps on other platforms
+     */
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class EmbedSetting {
+        private String height;
+        private String width;
+        private boolean showNavigationBar;
```

Comment:
For classes that are used to deserialize data from the database, consider using boxed versions of primary data types. Like `Boolean` instead of `boolean`.

Why? Because if this field's value is `null` in the MongoDB document, then the deserializer will attempt to set this field to `null`, and that will throw a `NullPointerException`, since variables with primary types can _never_ be set to `null`.

So, any class, if it will be used for any deserialization, it's best to just use boxed versions, and be cautious of `null` values.

```suggestion
        private Boolean showNavigationBar;
```

Reply:
Updated